.github/workflows/kiali-release.yml

```diff
@@ -0,0 +1,117 @@
+name: Kiali Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      release_type:
+        description: 'Release Type'     
+        required: true
+        default: 'auto' 
+        type: choice
+        options:
+        - auto
+        - minor
+        - snapshot.0
+        - snpashot.1
+        - edge 
+      backend_repo:
+        description: 'The GitHub repo of the back-end sources, in owner/repo format'
+        required: true
+        default: leandroberetta/kiali 
+        type: string
+      quay_name:
+        description: 'The name of the Quay repository to push the release'        
+        type: string
+        default: quay.io/leandroberetta/kiali
+        required: true 
+
+jobs:
+  # The initialize job gathers variables for later use in jobs.
+  # We are using this technique rather environment variables because at the moment, they won't work with reusable jobs.  
+  initialize:
+    name: Initialize
+    runs-on: ubuntu-latest
+    outputs:
+      target_branch: ${{ steps.target_branch.outputs.target_branch }}
+      release_type: ${{ steps.release_type.outputs.release_type || 'auto' }}      
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - id: target_branch
+        run: echo "::set-output name=target_branch::${{ github.base_ref || github.ref_name }}"        
+      - name: Determine release type
+        id: release_type
+        if: ${{ github.event.inputs.release_type == 'auto'}}
+        run: echo "::set-output name=release_type::$(./.github/workflows/util/release_type.sh)"
+  build_backend:
+    name: Build backend
+    uses: ./.github/workflows/build-backend.yml
+    needs: [initialize]
+  build_frontend:
+    name: Build frontend
+    uses: ./.github/workflows/build-frontend.yml
+    needs: [initialize]
+    with:
+      target_branch: ${{needs.initialize.outputs.target_branch}}
+  integration_tests_backend:
+    name: Run backend integration tests
+    uses: ./.github/workflows/integration-tests-backend.yml
+    needs: [initialize, build_backend, build_frontend]
+    with:
+      target_branch: ${{needs.initialize.outputs.target_branch}}
+  integration_tests_frontend:
+    name: Run frontend integration tests
+    uses: ./.github/workflows/integration-tests-frontend.yml
+    needs: [initialize, build_backend, build_frontend]
+    with:
+      target_branch: ${{needs.initialize.outputs.target_branch}}
+  release_kiali:
+    name: Release Kiali
+    runs-on: ubuntu-20.04
+    needs: [initialize, build_backend, build_frontend, integration_tests_backend, integration_tests_frontend]
+    env:
+      RELEASE_TYPE: ${{ needs.initialize.outputs.release_type}}        
+      BACKEND_REPO: ${{ github.event.inputs.backend_repo }}
+      QUAY_NAME: ${{ github.event.inputs.quay_name }}
+    steps:    
+      - name: Checkout code
+        uses: actions/checkout@v2
+      
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.17.7
```

Comment:
I wonder if it is possible to set the go version in a single place?
I see the go version spread over multiple workflow files.

Additionally, it is present on the main Makefile: https://github.com/kiali/kiali/blob/7d1c3350c142bd36c13245d7a90078d144faf7e9/Makefile#L31

It would be quite easy to forget to update all places when needed.